Masters.Api/Controllers/UserController.cs

```diff
@@ -1,25 +1,123 @@
 
-        [HttpPost("/Login")]
-        public IActionResult Login()
+        [HttpPost("[action]")]
+        public async Task<IActionResult> Register(RegisterBindingModel model)
         {
-            return Ok(authSettings.Value.Jwt.SecurityKey);
+            var user = new IdentityUser
+            {
+                Email = model.Email,
+                UserName = model.Email
+            };
+            var result = await userManager.CreateAsync(user, model.Password);
+            if (result.Succeeded)
+            {
+                var code = await userManager.GenerateEmailConfirmationTokenAsync(user);
+                code = WebEncoders.Base64UrlEncode(Encoding.UTF8.GetBytes(code));
+
+                return Ok(new { code = code, email = model.Email });
+            }
+            else
+            {
+                foreach (var error in result.Errors)
+                {
+                    ModelState.AddModelError(string.Empty, error.Description);
+                }
+                return BadRequest(ModelState);
+            }
+        }
+
+        [HttpPost("[action]")]
+        public async Task<IActionResult> Confirm(ConfirmBindingModel confirm)
+        {
+            var user = await userManager.FindByEmailAsync(confirm.Email);
+            var code = Encoding.UTF8.GetString(WebEncoders.Base64UrlDecode(confirm.Code));
+
+            if (user == null || user.EmailConfirmed)
+            {
+                return BadRequest();
+            }
+            else if ((await userManager.ConfirmEmailAsync(user, code)).Succeeded)
+            {
+                return Ok("Your email is confirmed");
+            }
+            else
+            {
+                return BadRequest();
+            }
         }
+
+        [HttpPost("login")]
+        public async Task<IActionResult> LoginAsync(LoginBindingModel login)
+        {
+            IActionResult actionResult;
+
+            var user = await userManager.FindByEmailAsync(login.Email);
+
+            if (user == null)
+            {
+                actionResult = NotFound(new { errors = new[] { $"User with email '{login.Email}' is not found" } });
+            }
+            else if (await userManager.CheckPasswordAsync(user, login.Password))
+            {
+                if (!user.EmailConfirmed)
+                {
+                    actionResult = BadRequest(new { errors = new[] { "Email is not confirmed. Please, go to your email account" } });
+                }
+                else
+                {
+                    var token = GenerateTokenAsync(user);
+                    actionResult = Ok(new { jwt = token });
+                }
+            }
+            else
+            {
+                actionResult = BadRequest(new { errors = new[] { "User password is not valid" } });
+            }
+
+            return actionResult;
+        }
+        private string GenerateTokenAsync(IdentityUser user)
+        {
+            IList<Claim> userClaims = new List<Claim>
```

Comment:
just some tip, you can actually store and retrieve user claims in identity i.e. userManager.AddClaimAsync to add claims during registration. No need to change this just additional info